source/extensions/filters/http/jwt_authn/authenticator.cc

```diff
@@ -247,9 +248,12 @@ void AuthenticatorImpl::verifyKey() {
 
   // Forward the payload
   const auto& provider = jwks_data_->getJwtProvider();
+  std::string payload_str_base64url_with_padding = jwt_->payload_str_base64url_;
+  Base64::completePadding(payload_str_base64url_with_padding);
```

Comment:
just call it "payload_with_padding"
move the code inside "if" so not to do the data copy when it is needed.

Reply:
It is only one copy here as I don't think it is worth polluting the `jwt` object if I modify/move `payload_str_base64url_`, though it is useless after this line.

Reply:
I am talking about the if in line 254.   If not need to add forward_payload_header,  we don't need to copy the payload, move the line 251,252 inside that "if".

Reply:
Done.